### PR TITLE
feat: Add new diffing

### PR DESCRIPTION
**Pull request by Khaledgarbaya**

### CODE REVIEW

File: lib/entry-state.js [hunk 1]
```diff
@@ -12,11 +13,32 @@ module.exports = async function attachEntryState (entry) {
...
+    if (isObject(field)) {
+      return [entry.fields[key], publishedEntry.fields[key]]
+    }
+  }).filter(Boolean)
+  entriesToCompare.forEach((item) => {
+    const originalItem = item[0]
+    const publishedItem = item[1]
+    if (isArray(originalItem)) {
+      originalItem.forEach((innerItem, index) => {
+        if (!isArray(innerItem) && isObject(innerItem) && (!innerItem.fields || originalItem[index].sys.updatedAt !== publishedItem[index].sys.updatedAt)) {
```
Comment (axe312ger) [inline lib/entry-state.js]: I miss a comment here what the long check actually does and why

File: lib/entry-state.js [hunk 2]
```diff
@@ -12,11 +13,32 @@ module.exports = async function attachEntryState (entry) {
...
+          entry.pendingChanges = true
+          return
+        }
+      })
+    } else if (isObject(item[0]) && (!item[0].fields || item[0].sys.updatedAt !== item[1].sys.updatedAt)) {
```
Comment (axe312ger) [inline lib/entry-state.js]: Same here, missing comment what and why

File: lib/entry-state.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
 const { getEntry } = require('./../services/contentful')
+const { isObject, isArray } = require('lodash')
 
 async function getPublishedEntry (entry) {
   try {
-    return await getEntry(entry.sys.id)
+    return await getEntry(entry.sys.id, entry.sys.contentType.sys.id)
```
Comment (axe312ger) [inline lib/entry-state.js]: For what do we need the content type here?

File: .gitignore [hunk 4]
```diff
@@ -65,3 +65,4 @@ cypress
 
 # e2e test directory
 test/e2e
+.vscode
```
Comment (axe312ger) [inline .gitignore]: +1 on `.vscode` but maybe we should straight use https://www.gitignore.io/api/vim%2Ccode%2Clinux%2Cmacos%2Cwindows%2Csublimetext%2Cnode?

File: lib/entry-state.js (hunk 3 above)
Comment (Khaledgarbaya) [reply lib/entry-state.js]: we are performing a query and in that case the `content_type` is required

File: .gitignore (hunk 4 above)
Comment (Khaledgarbaya) [reply .gitignore]: Somehow today I decided to use vs code. We can update that later

File: lib/entry-state.js (hunk 1 above)
Comment (Khaledgarbaya) [reply lib/entry-state.js]: 👍

File: lib/entry-state.js (hunk 2 above)
Comment (Khaledgarbaya) [reply lib/entry-state.js]: 👍

File: .gitignore (hunk 4 above)
Comment (axe312ger) [reply .gitignore]: Alright, created https://github.com/contentful/the-example-app.nodejs/issues/79

**Pull request merged**